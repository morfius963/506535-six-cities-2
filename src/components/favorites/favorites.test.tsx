import * as React from "react";
import * as renderer from "react-test-renderer";
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import Favorites from "./favorites";
import Footer from "../footer/footer";
import MainHeader from "../main-header/main-header";
import fixtureData from "../../__fixtures__/offers";

jest.mock(`../main-header/main-header`, () => jest.fn().mockReturnValue(null));
jest.mock(`../footer/footer`, () => jest.fn().mockReturnValue(null));
jest.mock(`../../store/actions/async-actions`, () => ({
  Operation: {}
}));

describe(`snapshot test`, () => {
  const appData = {
    favoriteOffers: fixtureData
  };

  const mockStore = configureStore([thunk]);
  const store = mockStore({appData});

  it(`Component correctly renders`, () => {
    const props = {
      onFavoriteCardToggle: jest.fn(),
      onFavoriteOffersLoad: jest.fn()
    };

    const tree = renderer
      .create(
          <Provider store={store}>
            <Favorites {...props} />
          </Provider>
      )
      .toJSON();

    expect(MainHeader).toHaveBeenCalled();
    expect(Footer).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
