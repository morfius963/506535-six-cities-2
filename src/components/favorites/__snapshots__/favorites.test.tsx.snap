// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<div
  className="page "
>
  <main
    className="page__main page__main--favorites "
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites "
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
